src/Discord/Parts/Interactions/Interaction.php

```diff
@@ -341,4 +366,150 @@ protected function respond(array $payload, ?Multipart $multipart = null): Extend
+     * @see updateOriginalResponse()
+     *
+     * @param string               $content          Content of the message.
+     * @param array[]|Embed[]|null $embeds           An array of up to 10 embeds. Can also be an array of DiscordPHP embeds.
+     * @param array|null           $allowed_mentions Allowed mentions object. See Discord developer docs.
+     *
+     * @return ExtendedPromiseInterface
+     */
+    public function updateInitialResponse(?string $content = null, ?array $embeds = null, ?array $allowed_mentions = null): ExtendedPromiseInterface
+    {
+        $builder = MessageBuilder::new()
+            ->setContent($content);
+
+        if ($embeds) {
+            $embeds = array_map(function ($e) {
+                if ($e instanceof Embed) {
+                    return $e->getRawAttributes();
+                }
+    
+                return $e;
+            }, $embeds);
+            $builder->setEmbeds($embeds);
+        }
+
+        if ($allowed_mentions) {
+            $builder->setAllowedMentions($allowed_mentions);
+        }
+
+        return $this->updateOriginalResponse($builder);
+    }
+
+    /**
+     * Deletes the original response to the interaction.
+     * Must have already used `reply` or `replyToSource`.
+     * @deprecated 7.1.0 Backported for DiscordPHP-Slash
+     * @see deleteOriginalResponse()
+     *
+     * @return ExtendedPromiseInterface
+     */
+    public function deleteInitialResponse(): ExtendedPromiseInterface
+    {
+        return $this->deleteOriginalResponse();
+    }
+
+    /**
+     * Updates a follow up message.
+     * Backported for DiscordPHP-Slash
+     *
+     * @param string               $message_id
+     * @param string|null          $content
+     * @param array[]|Embed[]|null $embeds
+     * @param array|null           $allowed_mentions
+     *
+     * @return ExtendedPromiseInterface
+     */
+    public function updateFollowUpMessage(string $message_id, string $content = null, array $embeds = null, array $allowed_mentions = null)
+    {
+        if ($this->type != Interaction::TYPE_APPLICATION_COMMAND) {
+            $this->discord->logger->warning('You can only update follow up messages that occur due to a application command interaction.');
+        }
+
+        if (! $this->responded) {
+            throw new RuntimeException('Cannot create a follow-up message as the interaction has not been responded to.');
+        }
+
+        return (function () use ($message_id, $content, $embeds, $allowed_mentions): ExtendedPromiseInterface {
+            return $this->http->patch(Endpoint::bind(Endpoint::INTERACTION_FOLLOW_UP, $this->application_id, $this->token, $message_id), [
+                'content' => $content,
+                'embeds' => $embeds,
+                'allowed_mentions' => $allowed_mentions,
+            ]);
+        })()->then(function ($response) {
+            return $this->factory->create(Message::class, $response, true);
+        });
+    }
```

Comment:
I'm not sure if we should backport these methods from the slash command client, it's already going to be a breaking change for users upgrading so perhaps it's just easier to forward them to the new APIs?

Reply:
I'm not sure either, it really depends on the current users of Slash command, aside of MessageBuilder, the method names can be confusing (e.g. updateInitialResponse vs updateOriginalResponse) so i choose to backport them and mark them as deprecated which will help them to move to the new API by time.
In my opinion this can be removed in future (e.g. discordphp v8?).

Reply:
I think it's best to get it over and done with now since it's a major conversion from a separate library. Functions that are missing can be ported over, but changed to use the new features such as `MessageBuilder` (if they are missing).

We can put together a conversion guide for users coming from the slash client which will make it easier.

Reply:
Done, but i keep the updateFollowUpMessage as this it is missing from here, however updated the methods to only accept MessageBuilder